### ISSUE TITLE
sokol_gfx.h: Problem with the GL state cache after deleting buffer

### ISSUE THREAD

**Issue by mgerdes** [floooh/sokol#360]
Hey,

I've been having an issue with deleting buffers. What happens is I create a buffer, bind the buffer, delete the buffer, then create a new buffer and I get a crash. I think the problem is that the state cache can be invalid after deleting a buffer.

According to this: https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glDeleteBuffers.xhtml
If glDeleteBuffers is called with the currently bound buffer, then the binding reverts to 0, but the state cache is not reverting the binding to 0 and I think this is causing my issue. To fix this I think the _sg_gl_destroy_buffer function should have code that looks something like this:

    if (_sg.gl.cache.vertex_buffer == buf->gl.buf[slot]) {
        _sg.gl.cache.vertex_buffer = 0;
    }

That works for me but I'm not sure if it's the right way to be handled. Or maybe I'm doing something else wrong?

Thanks for the great library! I've been really enjoying using it!

**Comment by floooh** [comment]
Yes this looks like a bug, the GL backend state cache's vertex- and index-buffer entry should definitely be invalidated when the a currently bound buffer is deleted, looks like the texture bindings have the same problem.

I'll try to come up with a more general solution (e.g. you also need to check the _sg.gl.cache.index_buffer entry).

I guess what happens is that the GL implementation reuses the same buffer name when you destroy and create a GL buffer (so when creating a new buffer you get back the same GL Id of the buffer you just destroyed). sokol_gfx.h would then think that the buffer binding is still correct (because the state cache hasn't been cleared), all the while the GL implementation has cleared the buffer binding.
